test/keyvault/7.2/azkeyvault/zz_generated_pollers.go

```diff
@@ -24,38 +25,53 @@ func (p *ClientFullBackupPoller) Done() bool {
 	return p.pt.Done()
 }
 
-// Poll fetches the latest state of the LRO.  It returns an HTTP response or error.
-// If the LRO has completed successfully, the poller's state is updated and the HTTP
+// Poll fetches the latest state of the LRO.
+// If the LRO has completed successfully, the poller's state is updated and the
 // response is returned.
 // If the LRO has completed with failure or was cancelled, the poller's state is
 // updated and the error is returned.
 // If the LRO has not reached a terminal state, the poller's state is updated and
-// the latest HTTP response is returned.
+// a zero-value response is returned.
```

Comment:
I envision the polling loop to look something like this.
```go
poller, err := op.BeginPostAsyncRetrySucceeded(context.Background(), nil)
handle(err)
var result LROsCustomHeaderClientPostAsyncRetrySucceededResponse
for !poller.Done() {
	result, err = poller.Poll(context.Background())
	handle(err)
	// sleep/do other things
}
fmt.Println(result)
```
With your suggestion the loop would look like this.
```go
poller, err := op.BeginPostAsyncRetrySucceeded(context.Background(), nil)
handle(err)
var result LROsCustomHeaderClientPostAsyncRetrySucceededResponse
for done := false; !done; done, err = poller.Poll(context.Background(), &result) {
	handle(err)
	// sleep/do other things
}
fmt.Println(result)
```
For the former, the pro I see is that its usage is somewhat similar to our pager pattern (I don't know how _big_ of a pro that is).  For the latter, the pro I see is that we can remove the `Done()` method at the expense of a slightly more verbose loop.

But generally speaking, you do raise a good point that `Poll()` only (obviously) communicates that it failed.  I guess we could leverage the zero-value to mean "not done" but that just seems weird.

Reply:
The fact that we went to lengths to remove the `Err()` method from our pagers due to the lack of clarity of the behavior seems to indicate that removing `Done()` would also be good (it feels very similar).

Reply:
Given that we consider `Poll()` to be the advanced scenario, I think it's ok if the polling loop is a bit more verbose.

Reply:
Another idea.  What if `Poll()` returned a pointer-to-type?  If the LRO is in a non-terminal state, it returns `nil, nil`.

Reply:
I like that better than returning a zero-value struct. I'm a little concerned that `nil, nil` won't be intuitive. It can look more like "nothing happened" than "the operation isn't complete". That's a getting started problem we can address in docs though.

Reply:
Thinking about this more, Poll() must return the `*http.Response` as it's the power-user case where complete control over the polling loop is required and the HTTP response is used to achieve that.  It also means we need a "final response" method, so I added `Result()`.